### ISSUE TITLE
update is not updating the modules

### ISSUE THREAD

**Issue by robsonpeixoto** [zimfw/zimfw#423]
In a terminal, run `zimfw info` and paste the output below:

```
zimfw version: 1.4.0 (built at 2021-01-07 18:39:43 UTC, previous commit is bcae8c0)
ZIM_HOME:      /Users/robinho/.zim
Zsh version:   5.8
System info:   Darwin updated6.local 20.3.0 Darwin Kernel Version 20.3.0: Thu Jan 21 00:07:06 PST 2021; root:xnu-7195.81.3~1/RELEASE_X86_64 x86_64
```

- gist of .zimrc: https://gist.github.com/robsonpeixoto/3a14f7e1e2df686dcc1ca5705503026e#file-zimrc
- gist of .zshrc: https://gist.github.com/robsonpeixoto/3a14f7e1e2df686dcc1ca5705503026e#file-zshrc

Description
-----------
`zimfw update` is not updating the repositories.

Steps to reproduce
------------------
```
❯ zimfw update -v
) termtitle: Already up to date.
) zsh-completions: Already up to date.
) utility: Already up to date.
) completion: Already up to date.
) ssh: Already up to date.
) environment: Already up to date.
) git: Already up to date.
) git-info: Already up to date.
) input: Already up to date.
) pure: Already up to date.
) zsh-syntax-highlighting: Already up to date.
) zsh-autosuggestions: Already up to date.
) zsh-history-substring-search: Already up to date.
Done with update. Restart your terminal for any changes to take effect.
) /Users/robinho/.zim/init.zsh: Already up to date
) /Users/robinho/.zim/login_init.zsh: Already up to date
Done with build.
Done with compile.


~/.zim/modules
❯ for app in *; do
… for ❯ cd $app
… for ❯ git pull --rebase
… for ❯ cd ..
… for ❯ done
Already up to date.
Already up to date.
Already up to date.
Already up to date.
Already up to date.
Already up to date.
Already up to date.
Already up to date.
Already up to date.
Already up to date.
Already up to date.
From github.com:zsh-users/zsh-completions
   244cbfc..c7baec4  master     -> origin/master
Updating 244cbfc..c7baec4
Fast-forward
 src/_cmake     |  2 +-
 src/_gpgconf   | 69 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 src/_nano      |  5 ++++
 src/_udisksctl |  2 +-
 4 files changed, 76 insertions(+), 2 deletions(-)
 create mode 100644 src/_gpgconf
Already up to date.
Already up to date.
```

**Comment by ericbn** [comment]
Oi @robsonpeixoto. Indeed, I was able to reproduce the issue. Thanks for reporting this.

This is the script that does the update: [src/tools/update.zsh.erb](https://github.com/zimfw/zimfw/blob/bcae8c0a06bbebf45e7ea06f623406d0804791e8/src/tools/update.zsh.erb). Looks like this is the commit that introduced the issue: https://github.com/zimfw/zimfw/commit/3567694a0f8f191cc50955151da0e38fd4f50711. The main change is that the default `REV` value in the script is `HEAD` instead of `master`. I thought `git fetch -pq origin HEAD` would fetch the repository default branch, but it seems that's not the case.

Just as a side note, we now have an [exa](https://github.com/zimfw/exa) module that does set pretty much the same aliases you have in your .zshrc.

**Comment by ericbn** [comment]
@robsonpeixoto, can up please test with https://github.com/zimfw/zimfw/blob/36aa1b3fe94bc79414a792f507f284be931504c2/zimfw.zsh ?

**Comment by robsonpeixoto** [comment]
Thanks @ericbn! Works like a charm!

```
~/.zim/modules/zsh-completions master
❯ git reset HEAD~5 --hard
HEAD is now at 9def41a Merge pull request #773 from simnalamburt/mssh

~/.zim/modules/pure main*
❯ git reset HEAD~5 --hard
HEAD is now at 8ec575c 1.15.0

❯ zimfw update
) exa: Already up to date.
) input: Already up to date.
) environment: Already up to date.
) utility: Already up to date.
) git-info: Already up to date.
) ssh: Already up to date.
) pure: Updating 8ec575c..5a43289
  * 5a43289 Remove antigen and antibody from instruction (#581) (2 days ago)
  * b83ad6d Update a link (4 weeks ago)
  * 5e315bc 1.16.0 (6 weeks ago)
  * 098df6b Improve performance of _is_inside_container check (#578) (6 weeks ago)
  * b6bc909 Fix readme typo (#579) (9 weeks ago)
) git: Already up to date.
) termtitle: Already up to date.
) archive: Already up to date.
) zsh-autosuggestions: Already up to date.
) completion: Already up to date.
) zsh-completions: Updating 9def41a..c7baec4
  *   c7baec4 Merge pull request #782 from davidhcefx/master (7 days ago)
  |\
  | * 37a9a48 update nano to version 5.4 (7 days ago)
  |/
  *   6fe9995 Merge pull request #781 from ScoreUnder/add-gpgconf (13 days ago)
  |\
  | * 4c5a8ef Add license notice to _gpgconf (13 days ago)
  | * ab80ddd Shorter perl one-liner in _gpgconf (13 days ago)
  | * a1d53f8 Add missing homedir option to _gpgconf (13 days ago)
  | * c90c27f Add completion for gpgconf (13 days ago)
  |/
  *   b8cc3b9 Merge pull request #777 from dancek/patch-1 (4 weeks ago)
  |\
  | * 19a4123 udisksctl: fix paths with a colon (4 weeks ago)
  |/
  *   00c7e28 Merge pull request #776 from askebm/cmake-targets-fix (5 weeks ago)
  |\
  | * 3511fb0 Use correct makefile for target completion (5 weeks ago)
  |/
  * 244cbfc Merge pull request #775 from brainplot/openssl-fix (6 weeks ago)
  * 17d92b0 Fix completion of cipher algorithms (6 weeks ago)
) zsh-syntax-highlighting: Already up to date.
) zsh-history-substring-search: Already up to date.
Done with update. Restart your terminal for any changes to take effect.
```

**Comment by ericbn** [comment]
Did a few more changes so it actually uses the default remote branch (first change was using the current local branch). Just released with version 1.4.1. Please let me know if you still face issues.
